### PR TITLE
Change slug pattern for numeric attribute values.

### PR DESCRIPTION
**Pull request by zedzior**
Issue: https://github.com/saleor/saleor/issues/15190

I want to merge this change, because it changes the slug of numeric attribute values from `{instance.id}_{attribute.id}` to `{actual_numeric_value}_{attribute.id}`. This will make numeric values reusable. Previous solution was resulting in redundant, duplicated attribute values created. 

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

File: saleor/graphql/attribute/utils.py [hunk 1]
```diff
@@ -605,10 +605,13 @@ def _pre_save_numeric_values(
         else:
             return tuple()
 
+        slug = slugify(unidecode(f"{value}_{attribute.id}"))
```
Comment (IKarbowiak) [inline saleor/graphql/attribute/utils.py:608]: It's not the right solution. The task what to investigate `why multiple rows are created` and not to change the number attribute slug pattern. 
We have two different types of attributes:
- `TYPES_WITH_CHOICES`
- `TYPES_WITH_UNIQUE_VALUES`
The numeric attribute is the type with unique values.

With your solution, two different instances of le's say `Product` may start sharing the `AttributeValue`. When one of these products is deleted, automatically `AttributeValue` is deleted, which means that we will end up with an empty value for attribute for the second `Product`.

**Review by IKarbowiak — CHANGES_REQUESTED** [review]

File: saleor/graphql/attribute/utils.py (hunk 1 above)
Comment (zedzior) [reply saleor/graphql/attribute/utils.py:608]: Hmm, are you sure that we delete shared attribute values? I wrote a new test, which shows, that with the solution:

1. we don't create duplicated attribute values anymore
2. after deleting one of the instances, which share the same attribute value, the value is still associated with remaining instances

Do I miss something? Can you please check it again?

Comment (IKarbowiak) [reply saleor/graphql/attribute/utils.py:608]: Here is the method that is responsible for that:
https://github.com/saleor/saleor/blob/main/saleor/graphql/product/mutations/product/product_delete.py#L85

**Comment by zedzior** [comment]
We decided to extend the solution for other unique attribute types.

**Pull request closed without merging**